### ISSUE TITLE
Move logic for handling partial api responses and errors to vets-api

### ISSUE THREAD

**Issue by JRRoof** [department-of-veterans-affairs/va.gov-team#99455]
## Description
Move any existing logic for handling partial api responses and errors in the appointments web application to the vets-api layer.

---

## Development Checklist

- [ ] Identify and document logic in the appointments web application
- [ ] Move existing logic to the vets-api layer
- [ ] Add additional parameters as needed
- [ ] Verify vets-api responses matches existing logic in appointments web application

### Design Assets
- N/A

### Developer Reference
N/A

## Definition of Done
- [ ] All tasks criteria are met
- [ ] [Technical Documentation]() is updated (must confirm if needed/or remove) 

---

**Comment by simiadebowale** [comment]
Thank you for looking into this, John. If I recall correctly, we’ve transitioned from handling individual error codes to using a more generic message for partial errors.

1. I believe that we should display the same message across both web and mobile. To support this, I’d suggest adding a field that includes both the error message title and body.(This way if we decide to make changes to this message in the future, there's one source of truth)
2. It would also be a good idea to create tickets to clean up the items you pointed out above.

**Comment by JunTaoLuo** [comment]
Update from standup. [Here](https://www.figma.com/design/eonNJsp57eqfPqx7ydsJY9/Feature-Reference-%7C-Appointments-FE?node-id=1387-31920&t=dFkkVSKysJoBelNt-0) are what the alerts look like currently, see the screenshots in the last column.

As for implementation, I think it might be easier for the backend to send formatted error messages for both appointment and requests in the response and for the frontend to decide which to display.

**Comment by simiadebowale** [comment]
I agree with the approach of including formatted error messages for both appointments and requests in the response, and allowing the frontend to determine which one to display.

**Comment by simiadebowale** [comment]
Ticket has been opened: https://github.com/orgs/department-of-veterans-affairs/projects/1554/views/6?pane=issue&itemId=104740770&issue=department-of-veterans-affairs%7Cva.gov-team%7C106625
